go/pkg/rexec/rexec.go

```diff
@@ -127,6 +127,32 @@ func (ec *Context) downloadOutputs(execRoot string) *command.Result {
 	return command.NewResultFromExitCode((int)(ec.resPb.ExitCode))
 }
 
+// HighAPIVersionNewerThanOrEqualTo returns whether the latest version reported
+// as supported in ServerCapabilities matches or is more recent than a
+// reference major/minor version.
+func highAPIVersionNewerThanOrEqualTo(serverCapabilities *repb.ServerCapabilities, major int32, minor int32) bool {
+	if serverCapabilities == nil {
+		return false
+	}
+
+	latestSupportedMajor := serverCapabilities.HighApiVersion.GetMajor()
+	latestSupportedMinor := serverCapabilities.HighApiVersion.GetMinor()
+
+	return (latestSupportedMajor > major || (latestSupportedMajor == major && latestSupportedMinor >= minor))
+}
+
+func actionHasPlatformProperties(serverCapabilities *repb.ServerCapabilities) bool {
```

Comment:
nit: I think all of this logic moved of the capabilities.go module, instead of exporting the ServerCaps. Then you can have client.ActionHasPublicProperties() and client.CommandHasOutputPaths() without arguments.

Reply:
I moved those methods to `Client` and figured they could use different names there: `Supports{ActionPlatformProperties, CommandOutputPaths}()`. What do you think?